azblob/zz_response_helpers.go

```diff
@@ -63,11 +63,9 @@ func (r *DownloadResponse) Body(o RetryReaderOptions) io.ReadCloser {
 	}
 	return NewRetryReader(r.ctx, r.Response(), r.getInfo, o,
 		func(ctx context.Context, getInfo HTTPGetterInfo) (*http.Response, error) {
-			resp, err := r.b.Download(ctx, getInfo.Offset, getInfo.Count,
-				BlobAccessConditions{
-					ModifiedAccessConditions: ModifiedAccessConditions{IfMatch: getInfo.ETag},
-				},
-				false)
+			resp, err := r.b.Download(ctx, getInfo.Offset, getInfo.Count, BlobAccessConditions{
+				ModifiedAccessConditions: ModifiedAccessConditions{IfMatch: getInfo.ETag},
+			}, false, ClientProvidedKeyOptions{})
```

Comment:
>ClientProvidedKeyOptions{} [](start = 13, length = 26)

What if the blob is encrypted? Don't we need to take the CPK info as an argument?

Reply:
Yes, that's correct. Thanks for pointing it out. This code snippet takes care of retrying download and would have failed with empty CPK. It was not failing earlier because all the blob downloads succeeded in the first attempt itself.